package_monitor/management/commands/refresh_packages.py

```diff
@@ -18,31 +18,41 @@
 logger = getLogger(__name__)
 
 
-def create_package_version(requirement: Requirement) -> None:
+def create_package_version(requirement: Requirement) -> PackageVersion:
     """Create a new PackageVersion from a requirement. Handles errors."""
     try:
-        PackageVersion(requirement=requirement).save()
+        pv = PackageVersion(requirement=requirement).save()
         logger.info("Package '%s' added.", requirement.name)  # noqa
     except IntegrityError:
         logger.info("Package '%s' already exists.", requirement.name)  # noqa
+        pv = PackageVersion.objects.get(package_name=requirement.name)
+    return pv
 
 
-def local() -> None:
+def local() -> List[PackageVersion]:
     """Load local requirements file."""
     logger.info("Loading requirements from local file.")
+    packages = []
     with open(REQUIREMENTS_FILE, "r") as f:
         requirements = parse(f)
         for r in requirements:
             logger.debug("Creating new package: %r", r)
-            create_package_version(r)
+            pv = create_package_version(r)
+            packages.append(pv)
 
+    return packages
 
-def remote() -> Dict[str, List]:
+
+def remote(packages: List[PackageVersion] = []) -> Dict[str, List]:
```

Comment:
Update to remove mutable default args (see https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments):

```suggestion
def remote(packages: List[PackageVersion] = None) -> Dict[str, List]:
```

Reply:
I had that originally; it causes lots of mypy problems, which requires adding `noqa` or worse voodoo to suppress.

Reply:
If you could, yes. In situations like this I always defer to "what would I think if this code was projected in size 128px font a conference" - and in this case I'd be more embarrassed about the mutable args than I would about `# type: ignore`.

Revisiting this after many years (we don't use it any more), this whole module needs a rethink - `remote` should just take a list of `PackageVersion`s, it shouldn't be optional at all. If you could fix the args issue I'll merge this and release a version. I can rework the implementation at some point in the future.

Reply:
fwiw, no need to rush out a new version.  I have other pending PRs to create, and they should be fine to squeeze into the current release.